text/0000-artifact-hub-integration.md

```diff
@@ -0,0 +1,339 @@
+maintainers.
+
+### [Artifact Kind](https://github.com/artifacthub/hub/blob/master/docs/repositories.md)
+
+OCI artifacts that Artifact Hub knows about. Artifact Hub supports over 20 kinds of Cloud Native artifacts, including
+Argo Templates, Backstage Plugins, Helm Charts, and others.
+
+# Motivation
+
+[motivation]: #motivation
+
+Component Buildpacks and Builders should be easily discoverable to make them more usable and approachable. Today, the
+[buildpacks registry](https://registry.buildpacks.io/) service is the solution to discoverability. This registry service
+is a very important tool, but there is room for improvement. There are many features that ArtifactHub.io provides that
+the buildpack registry does not provide. Some features include rich descriptions, usage information, activity,
+dependency information, email subscriptions to packages, webhooks, related packages, and, furthermore, it is a service
+that the buildpacks team does not need to maintain.
+
+CNB should consider utilizing this project instead of enhancing its own registry. In this way, the buildpack discovery
+service will benefit from more features and dedicated user experience enhancements. The CNB project can on rely on
+ArticactHub for this responsibility, rather than the CNB registry bespoke solution.
+
+Here is an [example of a helm chart](https://artifacthub.io/packages/helm/prometheus-community/prometheus) that is
+indexed on ArtifactHub.io. You can see here there are a lot of interesting features like security reports,
+vulnerabilities, usage stats, and all sorts of things.
+
+# What it is
+
+This RFC proposes changes that would affect both the Artifact Hub project and the CNB project.
+
+### Add support in Artifact Hub for component buildpacks and builders.
+
+Adding support in Artifact Hub is an independent operation. The Artifact Hub project could add support for buildpacks
+and builders without any integration on the CNB project side. This being said, adding support without a commitment on
+the CNB side could cause confusion to users because ArtifactHub.io could be interpreted as another official buildpack
+registry. This could also cause additional toil for buildpack authors if they are burdened with publishing to two
+indexes. Therefore, we should not do this unless we plan on migrating (in at least some capacity) to treat
+ArtifactHub.io as an official buildpacks registry. This work would happen in the Artifact Hub repository. Proposal
+discussion can be found [here](https://github.com/artifacthub/hub/issues/4352).
+
+### Integrate CNB services and tools to use ArtifactHub.io as the official CNB registry
+
+This should happen in phases so that buildpack authors and `pack` users have minimal disruption to established
+workflows. CNB is committed to backwards compatibility, but there are sometimes good reasons to break this backwards
+compatibility. This RFC proposes a slow phase-out and eventual decommission of the buildpacks registry. More details
```

Comment:
I don't think that shutting down the old buildpacks registry necessarily means breaking backwards compat. If ArtifactHub matches the features we provide today, then changing our "backend" means we're still backwards compat.

Reply:
It would be a bit different because the publishing process for the CNB registry is an explicit operation that happens against the cnb index repository. Future publishing would be implicit in that publishers modify a git repo they own and then the Artifact Hub scanner picks it up.